### DISCUSSION TITLE
Modbus interface - connecting additional devices?

### DISCUSSION THREAD

**Discussion by calum-mcfarlane** [nathanmarlor/foxess_modbus#406] [Q&A]
Has anyone else tried attaching an additional device (or even more than one) to their modbus bridge? I'm wondering if there is potential in this situation for the requests from this integration to collide with the "normal" manual modbus sensors I'd set up for the other device.

(Context is new heating system, which has a heat meter which can optionally be fitted with a Modbus RTU output module. 
I'm assuming that as long as I give the other device a different ID to my inverter they should be able to share the "bus", this is after all how modbus is *meant* to be able to work. The heat meter would be polled considerably less often than the inverter so hopefully the opportunities for collisions are minimal. Both RTU devices would need to support the same serial settings, happily the heat meter can support the 9600 baud, no parity, one stop bit settings that the inverter uses).

**Comment by canton7** [comment]
It *should* work fine? As you say, modbus is designed to work this way.

This only caveats I'd say are:

* If you're using a TCP to RS485 adapter (not ModbusTCP), then there's nothing to stop two TCP requests reaching the adapter at the same time (either from the same host or different hosts), and the adapter probably won't be able to sort it all out.
* If you're using a ModbusTCP adapter of questionable quality (e.g. the W610), and sending two requests at the same time (from the same host or different hosts), I've seen the responses get mixed. This is a case which the adapter *should* be able to handle fine, but some don't.

**Reply by calum-mcfarlane** [reply]
It'll either be the USR 410S or the Waveshare, both of which are ModbusTCP aware so hopefully it'll work fine. I'll report back any findings when I get there.
